### PR TITLE
Server-installation/Upgrading: add information about upgrading MongoDB

### PR DESCRIPTION
**Pull request by kjaskiewiczz**

### CODE REVIEW

File: 07.Server-installation/07.Upgrading/docs.md [hunk 1]
```diff
@@ -52,6 +52,19 @@ These can sometimes happen due to device decommissioning.
...
+Before upgrading Mender Server you must upgrade MongoDB from version 3.6 to version 4.4.
+It is not possible to upgrade MongoDB directly from version 3.6 to version 4.4.
+Instead, you should upgrade MongoDB from 3.6 to 4.0, then from 4.0 to 4.2
+and finally from 4.2 to 4.4.
+
+MongoDB provides instruction on how to perform the upgrades:
+* [upgrade MongoDB to 4.0](https://docs.mongodb.com/manual/release-notes/4.0/#upgrade-procedures)
+* [upgrade MongoDB to 4.2](https://docs.mongodb.com/manual/release-notes/4.2/#upgrade-procedures)
+* [upgrade MongoDB to 4.4](https://docs.mongodb.com/manual/release-notes/4.4/#upgrade-procedures)
+
```
Comment (mirzak) [inline 07.Server-installation/07.Upgrading/docs.md:67]: I, who know very little about MongoDB, have a very hard time following the linked instructions and how to translate them to a Mender server environment.

**Review by tranchitella — CHANGES_REQUESTED** [review]
I believe we have to provide a bit of guidance regarding the upgrade of MongoDB. At minimum, we should tell the user to manually bump the image tag of mongodb to the next version, start it using docker-compose, enter the container and upgrade the compatibility version using the admin command. This has to be repeated for each major version until it reaches MongoDB 4.4.

Can you please add commands and explanations on how to perform these upgrades?

**Comment by kjaskiewiczz** [comment]
@tranchitella @mirzak I'll provide a more detailed description

**Comment by kjaskiewiczz** [comment]
@tranchitella @mirzak I've updated the docs; it still needs some polishing so please wait with the review, but it would be great if you could just take a look and say if I wasn't too verbose

File: 07.Server-installation/07.Upgrading/docs.md [hunk 2]
```diff
@@ -52,6 +52,136 @@ These can sometimes happen due to device decommissioning.
...
+The MongoDB 3.6 instance must have `featureCompatibilityVersion` set to `3.6`.
+
+To check `featureCompatibilityVersion`:
+```
+db.adminCommand( { getParameter: 1, featureCompatibilityVersion: 1 } )
+```
+
+If you are using docker-compose based setup you can check it by running docker exec command:
+```bash
+docker exec -it menderproduction_mender-mongo_1 mongo ir --eval "db.adminCommand( { getParameter: 1, featureCompatibilityVersion: 1 } )"
```
Comment (tranchitella) [inline 07.Server-installation/07.Upgrading/docs.md:82]: What is the `ir` in the command line?

**Comment by mirzak** [comment]
> @tranchitella @mirzak I've updated the docs; it still needs some polishing so please wait with the review, but it would be great if you could just take a look and say if I wasn't too verbose

The approach looks good to me.

File: 07.Server-installation/07.Upgrading/docs.md (hunk 2 above)
Comment (kjaskiewiczz) [reply 07.Server-installation/07.Upgrading/docs.md]: name of the database on which I was experimenting with features incompatible across different versions of MongoDB;
I've already removed it; thanks for pointing this out; I was copy/pasting from the wrong terminal;

**Review by tranchitella — APPROVED** [review]
Great, well done!

**Review by mirzak — APPROVED** [review]
Looks good. I guess this needs to be cherry-picked to 2.5.x branch?

**Comment by kjaskiewiczz** [comment]
@mirzak yes, this should be in the 2.5.x for sure.
I was wondering if this change should be in the master branch?
If master is development then what upgrade really means? Upgrading from last stable to development? If that's the case then the last stable is 2.5 and the section about upgrading MongoDB shouldn't be there.

File: 07.Server-installation/07.Upgrading/docs.md [hunk 3]
```diff
@@ -52,6 +52,136 @@ These can sometimes happen due to device decommissioning.
 You can find instructions on how to clean up the deviceauth database
 in the [Troubleshoot](../../201.Troubleshoot/04.Mender-Server/docs.md) chapter.
 
+## Upgrading MongoDB
```
Comment (estenberg) [inline 07.Server-installation/07.Upgrading/docs.md:55]: Would rephrase to "Upgrade to MongoDB 4.4"

This is section is version-specific but merged to master, so we'll need to remove it in a while (maybe a year or so from now, or when we do the next major DB upgrade).

**Review by estenberg — APPROVED** [review]

**Comment by mirzak** [comment]
> @mirzak yes, this should be in the 2.5.x for sure.
> I was wondering if this change should be in the master branch?

I think I would also question if this really should be in master at all. Makes most sense if goes directly to 2.5.x.

**Comment by kjaskiewiczz** [comment]
@mirzak @estenberg change has been merged to 2.5.x and now I need a final decision if I should merge to master
in my opinion, we don't need it in master branch

**Comment by mirzak** [comment]
As this is not applicable in master, I would say that we close this one without merge

**Comment by estenberg** [comment]
:+1: makes sense to me as well, it should be a temporary version-specific problem only.
